devices/apa102/apa102_test.go

```diff
@@ -653,40 +653,35 @@ func BenchmarkWriteWhite(b *testing.B) {
 	}
 }
 
+func BenchmarkWriteWhite(b *testing.B) {
+	o := DefaultOpts
+	o.Intensity = 250
+	benchmarkWrite(b, o, 150*3, func(i int) byte { return 0xFF })
+}
+
 func BenchmarkWriteDim(b *testing.B) {
-	b.ReportAllocs()
-	pixels := [150 * 3]byte{}
-	for i := range pixels {
-		pixels[i] = 1
-	}
 	o := DefaultOpts
-	o.NumPixels = len(pixels) / 3
 	o.Intensity = 250
-	d, _ := New(spitest.NewRecordRaw(ioutil.Discard), &o)
-	_, _ = d.Write(pixels[:])
-	b.ResetTimer()
-	for i := 0; i < b.N; i++ {
-		_, _ = d.Write(pixels[:])
-	}
+	benchmarkWrite(b, o, 150*3, func(i int) byte { return 0x01 })
 }
 
 func BenchmarkWriteBlack(b *testing.B) {
-	b.ReportAllocs()
-	pixels := [150 * 3]byte{}
 	o := DefaultOpts
-	o.NumPixels = len(pixels) / 3
 	o.Intensity = 250
-	d, _ := New(spitest.NewRecordRaw(ioutil.Discard), &o)
-	_, _ = d.Write(pixels[:])
-	b.ResetTimer()
-	for i := 0; i < b.N; i++ {
-		_, _ = d.Write(pixels[:])
-	}
+	benchmarkWrite(b, o, 150*3, func(i int) byte { return 0x0 })
 }
 
 func BenchmarkWriteColorful(b *testing.B) {
+	o := DefaultOpts
+	o.Intensity = 250
+	o.Temperature = 5000
+	benchmarkWrite(b, o, 150*3, func(i int) byte { return uint8(i) + uint8(i>>8) })
+}
+
+func BenchmarkWriteColorfulVariation(b *testing.B) {
+	// Continuously vary the lookup tables.
 	b.ReportAllocs()
-	pixels := [150 * 3]byte{}
+	pixels := [256 * 3]byte{}
```

Comment:
Good point.

Reply:
I didn't actually change this -- the way it's presenting the diff is pretty confusing.  My goal was to keep the same benchmarks with the same data for each.